features/invite/impl/src/main/kotlin/io/element/android/features/invite/impl/response/AcceptDeclineInviteView.kt

```diff
@@ -45,13 +45,13 @@ fun AcceptDeclineInviteView(
             onErrorDismiss = {
                 state.eventSink(InternalAcceptDeclineInviteEvents.DismissDeclineError)
             },
-            confirmationDialog = {
-                val invite = state.invite.getOrNull()
-                if (invite != null) {
+            confirmationDialog = { confirming ->
+                // Note: confirming will always be of type ConfirmingDeclineInvite.
+                if (confirming is ConfirmingDeclineInvite) {
```

Comment:
Can't we just force cast it then?

Reply:
We could but why take the risk of a crash?
Previous code was not `val invite = state.invite!!`, so I prefer to keep the code safe.

Reply:
> We could but why take the risk of a crash?

Can we then log an error or upload some error report to sentry? I'm worried this might just silently fail, although the tests should cover it.

Reply:
My [previous attempt](https://github.com/element-hq/element-x-android/pull/3645/commits/40d82d62a1c06cfd35992a2efdc3e048867f89e4) to fix this would have avoid this drawback. But it was too invasive, so we go with this approach. This is not a new issue, so I guess this is fine.